Rework the global singleton section in the language reference

tronical opened the pull request:
Remove the explanation of using two-way bindings to expose globals.
While that still works of course, we don't need to mention it as the first example.
Instead provide inline examples with Rust and C++ how to access the globals directly.

The snippets will be hidden/shown by future markup, for now both are visible.

ogoffart commented on docs/langref.md:
```diff
@@ -813,6 +790,50 @@ import { Logic as MathLogic } from "math.slint";
 
+<details>
+<summary>Usage from Rust</summary>
+
```
A link to the ``[`slint::Global`]`` would be great as it has some example (and could get more) https://docs.rs/slint/latest/slint/trait.Global.html

tronical replied:
I can try to do that, but there are two problems:

(1) It will be a broken link in the C++ docs. That can be worked around with the hiding, but that's not approved yet.
(2) It will be a broken link when looking at the reference in GitHub.

If you don't mind that then I can go ahead and work more on this to add a link to the Rust global trait and the C++ global docs. Let me know :)

ogoffart commented on docs/langref.md:
````diff
@@ -768,33 +769,9 @@ Example := Rectangle {
 }
 ```
 
-It is possible to re-expose a callback or properties from a global using the two way binding syntax.
````
I think this should stay in this reference.

> While that still works of course, we don't need to mention it as the first example.

But this page is a reference and should document the whole language entirely.

I guess now this page serves both as a reference and as a introduction to the language and that's maybe a mistake.
Maybe the other reciepes you are working on vould be used as an introduction to the language instead.

tronical replied:
Maybe not quite "introduction" but any example we provide (including in which order) becomes a "best practice".

If you'd like to retain this in the reference, then I can add a sentence to the two-way bindings instead?

tronical commented:
I've rebased this. Instead of removing the two-way binding section, I've just moved it downwards. I've also added the tags to enable automatic showing/hiding of the snippets depending on whether it's Rust or C++.
